core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaUpdateRecipesTranslator.java

```diff
@@ -337,6 +403,14 @@ private ItemDescriptorWithCount[][] combinations(GeyserSession session, Ingredie
         return combinations;
     }
 
+    private void addRecipeIdentifier(GeyserSession session, String javaIdentifier, List<String> bedrockIdentifiers) {
+        if (session.getJavaToBedrockRecipeIds().containsKey(javaIdentifier)) {
```

Comment:
This should be either replace the containsKey check with get and null check, or use `computeIfAbsent`.